Sources/SnapshotTesting/Snapshotting/NSView.swift

```diff
@@ -9,17 +9,48 @@ extension Snapshotting where Value == NSView, Format == NSImage {
 
   /// A snapshot strategy for comparing views based on pixel equality.
   ///
+  /// >This function calls to `NSView.cacheDisplay()` which has side-effects that cannot be undone. Under some circumstances
+  /// >subviews will be added (e.g. for `NSButton`-views) and `NSView.needsLayout` will be set to `false`. Keep that in mind
+  /// >when asserting with `.image` and `.recursiveDescription` within the same test.
+  ///
   /// - Parameters:
   ///   - precision: The percentage of pixels that must match.
   ///   - perceptualPrecision: The percentage a pixel must match the source pixel to be considered a match. [98-99% mimics the precision of the human eye.](http://zschuessler.github.io/DeltaE/learn/#toc-defining-delta-e)
   ///   - size: A view size override.
-  public static func image(precision: Float = 1, perceptualPrecision: Float = 1, size: CGSize? = nil) -> Snapshotting {
+  ///   - appearance: The appearance to use when drawing the view. Pass `nil` to use the view’s existing appearance.
+  ///   - windowForDrawing: The choice of window to use when drawing the view. Pass `nil` to ignore.
+  public static func image(
+    precision: Float = 1,
+    perceptualPrecision: Float = 1,
+    size: CGSize? = nil,
+    appearance: NSAppearance? = NSAppearance(named: .aqua),
+    windowForDrawing: GenericWindow? = nil
```

Comment:
A few questions about this.

- Any reason why this is `nil` by default and not a `GenericWindow()` that produces more deterministic snapshots? Is it just to keep compatibility with existing snaps? We're prepping a release soon that has other bug fixes that may break some existing snapshots, so we are OK making it quite clear in the release notes that upon upgrading, folks should regenerate and audit their snapshot suites. 
- Is it possible to make `GenericWindow` a private concern of the library and instead introduce options for `backingScaleFactor` and `colorSpace`? Would we be missing out on other options?
- Alternately, could this be a more generic `NSWindow` and then we could default it to be the deterministic 2x scale and RGB color space?